### DISCUSSION TITLE
USB HID controls

### DISCUSSION THREAD

**Discussion by HengYeDev** [perpetualKid/FreeTrainSimulator#74] [Ideas]
It could be possible in the future to enable support for USB HID controls such as flight sim throttles.

**Comment by perpetualKid** [comment]
indeed, this is what the work on #47 was about, to implement some framework to allow differnt kind of input mapped to actions in the program, to handle and control train behavior. Keyboard and Mouse input are not refactored to this new approach, and #70 is work in progress for [RailDriver](https://www.raildriver.com) input. I'm tempted to implement Gamepad support as well, but this will be lower priority. 
Other custom/proprietary controls could be implemented following same pattern if there is a need or someone picking up that task.

I think this is more appropriate in discussions for now as this is not an open issue really?

**Reply by HengYeDev** [reply]
i was thinking that feature requests go in issue tracker too like some other projects....

**Reply by perpetualKid** [reply]
I see. I think it needs some backlog for planned or wanted feastures, but not necessarily an open issue until it gets as work in progress?

As for the controller, do you have some particular device in mind? If we'd have generic gamepad, and RailDriver, it should cover many use cases. The challenge is that non-Key input needs some mapping to controls, and since Monogame isn't particularly abstracting game input, this may need separate implementation for each kind.

**Reply by HengYeDev** [reply]
it can have an open issue with a feature-request tag or similar. putting in community makes it messy

Most devices are generic gamepad but do need mapping...maybe a mapping dialog
